utils/vqm2blif/src/base/cleanup.cpp

```diff
@@ -848,8 +836,64 @@ void remove_node ( t_node* node, t_node** nodes, int original_num_nodes ) {
 	}
 	
 	VTR_ASSERT(found);
-	oneluts_elim++;
 }
 
 //============================================================================================
 //============================================================================================
+
+void reorganize_module_node_list(int original_number_of_nodes, int number_of_nodes_eliminated, t_node** module_node_list, t_module* module)
+{
+	/*
+        The purpose of this function is to regorganize the nodes array by filling in gaps with the last available elements in the array to save CPU time.
+
+		The node array provided to this function is expected to have elements deleted within. THis essentially creates gaps in the array. So we fill in the gaps of this array and ensure it is a continuous list of nodes.
+
+		Please refer to the example below:
+
+		Inital Node array:
+             ------      ------                  ------      ------
+            |LUT 1| --> |LUT 2| -->         --> |LUT 3| --> |LUT 4|
+            ------      ------                  ------      ------
+
+		After reorganization and filling gaps:
+
+             ------      ------      ------      ------
+            |LUT 1| --> |LUT 2| --> |LUT 3| --> |LUT 4| -->
+            ------      ------      ------      ------
+
+	Change Log:
+		- Srivatsan Srinivasan, August 2021:
+			- created this function to reorganize node arrays with gaps inside of them.
+			- Initially the feature provided by this function was embedded indide the "remove_one_lut_nodes" function. By creating a seperate function, we are now not restricted to only removing one-lut nodes. 
+			- Now we can remove any types of nodes and then run this function to reorganize the node array.  
+*/
+	int new_array_size = original_number_of_nodes - number_of_nodes_eliminated;
+	int curr_node_index = 0;
+	int replacement_node_index = original_number_of_nodes - 1;
+	while (curr_node_index < replacement_node_index) {
+		if (module_node_list[curr_node_index] == NULL) {
+			if (module_node_list[replacement_node_index] != NULL) {
+				//Replace gap with node
+				module_node_list[curr_node_index] = module_node_list[replacement_node_index];
+				module_node_list[replacement_node_index] = NULL;
+				curr_node_index++;
+			}
+			replacement_node_index--;
+		} else {
+			curr_node_index++;
+		}
+	}
+	if (module_node_list[curr_node_index] == NULL) {
+		VTR_ASSERT(curr_node_index == new_array_size); //check array size
```

Comment:
`// check array size` is too low level a comment (doesn't really tell me what you're doing except that it is a check, which I can already tell from the assert.
Instead, comment the whole if statement to explain what you're checking, and why you need two cases.

Reply:
This entire block has now been removed. Originally, this assert block was used to verify whether the node list had the correct number of nodes after we had filled in all the gaps. The user originally supplied a parameter of the number of nodes within the array that were removed, using this a new array size was determined and this was compared to size of the node array after filling in all the gaps. But now, since we do not pass in the number of nodes removed, there is no need to perform this check as we now internally determine the node array size as we fix it.